index.js

```diff
@@ -10,8 +10,13 @@
 var isUTF8 = require('is-utf8');
 var isBuffer = require('is-buffer');
 
+function maybeUTF8(buf) {
```

Comment:
I was just thinking that if you're doing this as an optimization then you probably don't need the extra function and a comment should suffice. Either way, I think it's fine to do the slice and if you want to squeeze out a little more performance, you could remove the function.